src/test/java/org/schemaspy/connection/WithPasswordTest.java

```diff
@@ -0,0 +1,29 @@
+package org.schemaspy.connection;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+/**
+ * Tests for {@link WithPassword}.
+ */
+public final class WithPasswordTest {
+
+    /**
+     * Given a password and an origin,
+     * When the object is asked for connection properties,
+     * Then its response should include the password.
+     */
+    @Test
+    public void parse() throws IOException {
```

Comment:
Parse feels strange as testname. Can we come up with a better one?

Reply:
I'm open for suggestions :slightly_smiling_face: 
If it helps, I mentally think "I can [test method name]"

Reply:
I can SetAnActualPassword

Reply:
`readPasswords`, maybe? But I still think "parse" is clearer

Reply:
It doesn't read, it doesn't parse and it's a with password so something gets a password. Hence a password is set not in the setter way.

Reply:
Hmm, I think we're doing two (three, actually) things. The first one should probably be decomposed in a future refactoring.

1. read/parse the password from config
2. decorate origin with the password
3. `null` handling

I think you saw it from (2.) while I was focusing on (1.)

"I can decorate"?
"I can supply password"?

Reply:
It's not reading or parsing from config, that might be something that config does. At most we retrieve it from config.

Yes decorating, in a sense, it feels more like a chained command. 

Null handling is a different test case. We should only handle a single code path in a single unit test. But that's more of a glass half full or half empty.

Do we handle null or do we handle non-null?